### PR TITLE
Add .NET Standard 2.0 support to serialization & core abstraction libraries

### PR DESCRIPTION
**Pull request by ReubenBond**
This commit adds .NET Standard 2.0 support to:

* `Microsoft.Orleans.Serialization.Abstractions`
* `Microsoft.Orleans.Serialization`
* `Microsoft.Orleans.Serialization.FSharp`
* `Microsoft.Orleans.Serialization.TestKit`

It's possible we could add .NET Standard support elsewhere. I'm wary of that, since it does decidedly hamper us.

### CODE REVIEW

**Comment by pentp** [comment]
Core.Abstractions should probably not depend on Serialization (only Serialization.Abstractions). I tried to move some of the stuff for a quick proof of concept, but the dependencies are quite entangled there... This is a separate issue from this PR, but somewhat related.

In general I think it doesn't make sense to add netstandard2.0 to anything other than the Abstractions packages.

File: Directory.Build.props [hunk 1]
```diff
@@ -42,13 +42,18 @@
   <!-- Shared Package Versions -->
   <PropertyGroup>
     <!-- System packages -->
+    <SystemCollectionsImmutableVersion>5.0.0</SystemCollectionsImmutableVersion>
```
Comment (pentp) [inline Directory.Build.props:45]: Could move these to a separate block with a comment about these being for only netstandard2.0 (like it was before my last major cleanup that removed these).

File: src/Orleans.Transactions.Abstractions/Orleans.Transactions.Abstractions.csproj [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <PackageTags>$(PackageTags) Transactions</PackageTags>
+    <TargetFrameworks>$(StandardTargetFrameworks)</TargetFrameworks>
+    <AssemblyName>Orleans.Transactions.Abstractions</AssemblyName>
+    <RootNamespace>Orleans.Transactions</RootNamespace>
+    <OrleansBuildTimeCodeGen>true</OrleansBuildTimeCodeGen>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Orleans.Core.Abstractions\Orleans.Core.Abstractions.csproj" />
+    <ProjectReference Include="..\Orleans.Serialization.Abstractions\Orleans.Serialization.Abstractions.csproj" />
```
Comment (pentp) [inline src/Orleans.Transactions.Abstractions/Orleans.Transactions.Abstractions.csproj:15]: These two references aren't needed, `Orleans.Core.Abstractions` should be enough.

**Review by pentp — APPROVED** [review]

**Comment by ReubenBond** [comment]
I wonder why this is failing. If anyone has any idea, I'd love to hear it

**Pull request closed without merging**